### PR TITLE
initial skeleton of vector tiles prototype implementation

### PR DESCRIPTION
**Pull request by iverase**
Initial commit of a new end point that generates vector tiles from geo spatial data stored in Elasticsearch. It adds three packages in the spatial module:

- org.elasticsearch.xpack.spatial.vectortile: Contains the classes that introduces the new end point.
- org.elasticsearch.xpack.spatial.vectortile.collector: contains different implementation of how to construct the vector tile. 
~~- org.elasticsearch.xpack.spatial.proto: Google protobuffer classes for generating vector tiles.~~

The new end-point can be used by calling:

```
GET {index}/_mvt/{geo_field}/{z}/{x}/{y}
```

### CODE REVIEW

File: x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/proto/VectorTile.java [hunk 1]
```diff
@@ -0,0 +1,5846 @@
...
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// Generated by the protocol buffer compiler.  DO NOT EDIT!
+// source: org/elasticsearch/xpack/spatial/vector_tile.proto
+
+package org.elasticsearch.xpack.spatial.proto;
+
+public final class VectorTile {
```
Comment (nik9000) [inline x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/proto/VectorTile.java:13]: Is there any chance we could generate this as part of the build and land it into a "generated sources" location?

Comment (talevy) [reply x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/proto/VectorTile.java:13]: +1

Comment (iverase) [reply x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/proto/VectorTile.java]: Right. What I did is to remove the prototype package all together. Instead I am using the generated files in the map box_vector_tile dependency.

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/VectorTileAggregatorTests.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            }
+            if (randomBoolean()) {
+                w.forceMerge(1);
+            }
+            assertVectorTile(w, 0, 0, numDocs, new GeoPointFieldMapper.GeoPointFieldType("field"));
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testGeoShape() throws Exception {
```
Comment (imotov) [inline x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/VectorTileAggregatorTests.java:85]:
```suggestion
    @AwaitsFix(bugUrl = "That doesn't work yet")
    public void testGeoShape() throws Exception {
```

**Review by imotov — APPROVED** [review]
Let's suppress the broken test and merge this into the feature branch. There are several things that we should fix, but it will be easier to work no it in the branch.

**Comment by iverase** [comment]
@elasticmachine update branch

**Pull request merged**